packages/nocodb/src/db/BaseModelSqlv2.ts

```diff
@@ -1600,6 +1612,45 @@ class BaseModelSqlv2 {
       if (!checkColumnRequired(column, fields, extractPkAndPv)) continue;
 
       switch (column.uidt) {
+        case UITypes.DateTime:
+          if (this.isMySQL) {
+            // MySQL stores timestamp in UTC but display in timezone
+            // To verify the timezone, run `SELECT @@global.time_zone, @@session.time_zone;`
+            // If it's SYSTEM, then the timezone is read from the configuration file
+            // if a timezone is set in a DB, the retrieved value would be converted to the corresponding timezone
+            // for example, let's say the global timezone is +08:00 in DB
+            // the value 2023-01-01 10:00:00 (UTC) would display as 2023-01-01 18:00:00 (UTC+8)
+            // our existing logic is based on UTC, during the query, we need to take the UTC value
+            // hence, we use CONVERT_TZ to convert back to UTC value
+            res[sanitize(column.title || column.column_name)] =
+              this.dbDriver.raw(
+                `CONVERT_TZ(??, @@GLOBAL.time_zone, '+00:00')`,
+                [
+                  `${sanitize(alias || this.model.table_name)}.${
+                    column.column_name
+                  }`,
+                ],
+              );
+            break;
+          } else if (this.isPg) {
+            // if there is no timezone info, convert it to UTC
+            if (column.dt !== 'timestamp with time zone') {
```

Comment:
You can include `timestamptz` as well which is an alias for `timestamp with time zone`.